d3a_interface/data_classes.py

```diff
@@ -0,0 +1,442 @@
+                         original_price=original_price,
+                         attributes=attributes, requirements=requirements)
+        self.seller = seller
+        self.seller_origin = seller_origin
+        self.seller_origin_id = seller_origin_id
+        self.seller_id = seller_id
+
+    def __hash__(self) -> int:
+        return hash(self.id)
+
+    def __repr__(self) -> str:
+        return ("<Offer('{s.id!s:.6s}', '{s.energy} kWh@{s.price}', '{s.seller} {rate}'>"
+                .format(s=self, rate=self.energy_rate))
+
+    def __str__(self) -> str:
+        return ("{{{s.id!s:.6s}}} [origin: {s.seller_origin}] "
+                "[{s.seller}]: {s.energy} kWh @ {s.price} @ {rate}"
+                .format(s=self, rate=self.energy_rate))
+
+    def serializable_dict(self) -> Dict:
+        return {**super().serializable_dict(),
+                "seller": self.seller,
+                "seller_origin": self.seller_origin,
+                "seller_origin_id": self.seller_origin_id,
+                "seller_id": self.seller_id,
+                }
+
+    def __eq__(self, other) -> bool:
+        return (self.id == other.id and
+                self.price == other.price and
+                self.original_price == other.original_price and
+                self.energy == other.energy and
+                self.seller == other.seller and
+                self.seller_origin_id == other.seller_origin_id and
+                self.attributes == other.attributes and
+                self.requirements == other.requirements)
+
+    def csv_values(self) -> Tuple:
+        rate = round(self.energy_rate, 4)
+        return rate, self.energy, self.price, self.seller
+
+    @classmethod
+    def csv_fields(cls):
+        return "rate [ct./kWh]", "energy [kWh]", "price [ct.]", "seller"
+
+    @staticmethod
+    def copy(offer: "Offer") -> "Offer":
+        return Offer(offer.id, offer.time, offer.price, offer.energy, offer.seller,
+                     offer.original_price, offer.seller_origin, offer.seller_origin_id,
+                     offer.seller_id, attributes=offer.attributes, requirements=offer.requirements)
+
+
+class Bid(BaseBidOffer):
+    def __init__(self, id: str, time: datetime, price: float,
+                 energy: float, buyer: str,
+                 original_price: Optional[float] = None,
+                 buyer_origin: str = None,
+                 buyer_origin_id: str = None,
+                 buyer_id: str = None,
+                 attributes: Dict = None,
+                 requirements: List[Dict] = None
+                 ):
+        super().__init__(id=id, time=time, price=price, energy=energy,
+                         original_price=original_price,
+                         attributes=attributes, requirements=requirements)
+        self.buyer = buyer
+        self.buyer_origin = buyer_origin
+        self.buyer_origin_id = buyer_origin_id
+        self.buyer_id = buyer_id
+
+    def __hash__(self) -> int:
```

Comment:
Where do you need it?

Reply:
This can be useful when trying to set an offer/bid or any other market structure as a dictionary key, it will know how to hash the object

Reply:
Please move it to the base class, this applies to all child classes.

Reply:
@spyrostz for some reason when I implement __hash__() in the parent class, the children are not inheriting it, I'm still investigating why 
But it's probably due to the fact the that we're overriding the __eq__ in children classes https://stackoverflow.com/a/2134487

Reply:
I see, in this case LGTM